app/services/ui_notifications/hosts/destroy.rb

```diff
@@ -10,12 +10,21 @@ def create
         Notification.create!(
           initiator: initiator,
           audience: audience,
-          # note we do not store the subject, as the object is being deleted.
+          subject: subject.owner, # note we store the host owner, as the host object is being deleted.
           message: StringParser.new(blueprint.message, {subject: subject}),
           notification_blueprint: blueprint
         )
       end
 
+      def audience
+        case subject.owner_type
```

Comment:
is destroy the only action this should exists at?

Reply:
yes, because for destroy the subject is not passed to the Notification model as the subject is being deleted.